autoscheduler/frontend/src/tests/CourseSelectCard.test.tsx

```diff
@@ -9,98 +14,173 @@ import { Provider } from 'react-redux';
+      } = render(
+        <Provider store={store}><CourseSelectCard id={0} /></Provider>,
+      );
+
+      // act
+      // fill in course
+      const courseEntry = getByLabelText('Course') as HTMLInputElement;
+      act(() => { fireEvent.change(courseEntry, { target: { value: 'CSCE ' } }); });
+      const csce121Btn = await waitForElement(() => getByText('CSCE 121'));
+      act(() => { fireEvent.click(csce121Btn); });
+
+      // switch to section view
+      act(() => { fireEvent.click(getByText('Section')); });
+      const course1Sections = (await waitForElement(() => getAllByText(/50\d/))).length;
+
+      // change course and read sections again
+      act(() => { fireEvent.change(courseEntry, { target: { value: 'MATH 15' } }); });
+      const math151Btn = await waitForElement(() => getByText('MATH 151'));
+      act(() => { fireEvent.click(math151Btn); });
+      const course2Sections = (await waitForElement(() => getAllByText(/51\d/))).length;
+
+      // assert
+      expect(course1Sections).not.toEqual(0);
+      expect(course2Sections).not.toEqual(0);
+      expect(course1Sections).not.toEqual(course2Sections);
+    });
+  });
+
+  describe('only fetches api/sections once', () => {
+    describe('when we search and go to the Sections tab', () => {
+      test('and collapse then expand the card', async () => {
+        // arrange
+        fetchMock.mockResponseOnce(JSON.stringify({
+          results: ['CSCE 121'],
+        }));
+        fetchMock.mockImplementationOnce(testFetch); // For api/sections
+
+        const nodeProps = Object.create(Node.prototype, {});
+        // @ts-ignore
+        document.createRange = (): Range => ({
+          setStart: (): void => {},
+          setEnd: (): void => {},
+          commonAncestorContainer: {
+            ...nodeProps,
+            nodeName: 'BODY',
+            ownerDocument: document,
+          },
+        });
+        const store = createStore(autoSchedulerReducer, applyMiddleware(thunk));
+        const { getByText, getByLabelText } = render(
+          <Provider store={store}><CourseSelectCard id={0} /></Provider>,
+        );
+
+        // fill in course so we can show the sections
+        const courseEntry = getByLabelText('Course') as HTMLInputElement;
+        act(() => { fireEvent.change(courseEntry, { target: { value: 'CSCE ' } }); });
+        const csce121Btn = await waitForElement(() => getByText('CSCE 121'));
+        act(() => { fireEvent.click(csce121Btn); });
+
+        // act
+        // switch to sections view
+        act(() => { fireEvent.click(getByText('Section')); }); // Makes api/sections be called
+
+        // collapse then expand the card
+        act(() => { fireEvent.click(getByText('Collapse')); });
+        act(() => { fireEvent.click(getByLabelText('Expand')); });
+
+        // assert
+        // Need to assert that api/sections was only called once somehow?
+        //
+        // expect(fetchMock).toBeCalledTimes(1);
```

Comment:
We have a couple of options with how we can do this. The simplest is to assert `expect(fetchMock.mock.calls).toHaveLength(2)` so that it's called once for search and once for sections. 

Alternatively, if we swap out the `jest-fetch-mock` library for the `fetch-mock-jest` library, we can assert more idiomatically `expect(fetchMock).toHaveFetchedTimes(2)`. Beware that with either of these, you will have to call `fetchMock.mockClear()` at the top of the arrange phase for this test to erase data from other tests.

In addition to the choice between the two libraries, we can also choose to assert only on the calls made to `/api/sections` (sort of). To do this, we have to make a custom implementation like so:

```
fetchMock.mockImplementation((route: string) => {
          if (route.match(/api\/course.+/)) {
            return Promise.resolve(new Response(JSON.stringify({
              results: ['CSCE 121'],
            })));
          }
          if (route.match(/api\/sections.+/)) {
            sectionCalls += 1;
            return Promise.resolve(new Response('[]'));
          }
          return Promise.resolve(new Response('404 Not Found'));
        });
```
This works by making a custom variable called `sectionCalls` that is incremented whenever the URL matches api/section/whatever. We can then assert on `sectionCalls`.

Reply:
The fact that there's a `jest-fetch-mock` and `fetch-mock-jest` makes me laugh

Reply:
I'd say I like the third one the best so that's what I'm gonna go with